include/BSMPT/utility/utility.h

```diff
@@ -67,6 +108,138 @@ std::ostream &operator<<(std::ostream &os, const std::vector<T> &vec)
   return os;
 }
 
+/**
+ * @brief vector addition
+ */
+template <typename T>
+std::vector<T> operator+(const std::vector<T> &a, const std::vector<T> &b)
+{
+  assert(a.size() == b.size());
```

Comment:
assert does nothing in release mode. Why not throw an exception instead?

Reply:
Fixed.